diff --git a/dist/compiler/server/write.js b/dist/compiler/server/write.js
index 56963c0c0d728f6697aba12d31a13411b825a5da..8620b85daf55d2d322d480f7eef19dcc51983de8 100644
--- a/dist/compiler/server/write.js
+++ b/dist/compiler/server/write.js
@@ -42,7 +42,7 @@ async function write(config, outputFiles) {
     if (file.path.endsWith(".js") || file.path.endsWith(".mjs")) {
       // fix sourceMappingURL to be relative to current path instead of /build
       let filename = file.path.substring(file.path.lastIndexOf(path__namespace.sep) + 1);
-      let escapedFilename = filename.replace(/\./g, "\\.");
+      let escapedFilename = filename.replace(/([.\[\]])/g, '\\$1');
       let pattern = `(//# sourceMappingURL=)(.*)${escapedFilename}`;
       let contents = Buffer.from(file.contents).toString("utf-8");
       contents = contents.replace(new RegExp(pattern), `$1${filename}`);